hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/serialization/MarkerWrapper.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.hibernate.serialization;
+
+
+import org.hibernate.cache.access.SoftLock;
+
+/**
+ * A wrapper class for ExpiryMarker for returning copy of marker object with SoftLock
+ * marker interface.
+ */
+
+public class MarkerWrapper extends ExpiryMarker implements SoftLock {
```

Comment:
Does this need to extend `ExpiryMarker`? Can it not use composition to include the marker?

```
public class MarkerWrapper implements SoftLock {
    private final ExpiryMarker marker;
    public ExpiryMarker getMarker() {
        return marker;
    }
} 
```

This avoids the need for all the setters on `ExpiryMarker` which I believe complicates reasoning on the state of `ExpiryMarker`. The original idea was that `ExpiryMarker` was effectively immutable, it only had immutable fields for serialization/deserialization.